modules/gapi/src/backends/ov/govbackend.cpp

```diff
@@ -838,24 +924,35 @@ class PrePostProcWrapper {
         const auto &input_shape = m_model->input(input_name).get_shape();
         auto &input_info = m_ppp.input(input_name);
 
-        m_ppp.input(input_name).tensor().set_element_type(toOV(matdesc.depth));
-        if (isImage(matdesc, input_shape)) {
+        auto isMat = cv::util::holds_alternative<cv::GMatDesc>(input_meta);
+        auto prec  = isMat ? cv::util::get<cv::GMatDesc>(input_meta).depth : CV_8U;
+        m_ppp.input(input_name).tensor().set_element_type(toOV(prec));
+
+        const auto &matdesc   = isMat ? cv::util::get<cv::GMatDesc>(input_meta) : cv::GMatDesc();
```

Comment:
Instead of `matdesc` and `framedesc` you can simply do:

```
<< " is not compatible with input data " << isMat ? cv::util::get<GMatDesc>(input_meta) : cv::util::get<GFrameDesc>(input_meta) << ...
```